src/hotspot/share/logging/logTagSet.cpp

```diff
@@ -124,16 +124,31 @@ void LogTagSet::vwrite(LogLevelType level, const char* fmt, va_list args) {
     ret = os::vsnprintf(buf, sizeof(buf), fmt, args);
   }
   assert(ret >= 0, "Log message buffer issue");
-  if ((size_t)ret >= sizeof(buf)) {
-    size_t newbuf_len = prefix_len + ret + 1;
-    char* newbuf = NEW_C_HEAP_ARRAY(char, newbuf_len, mtLogging);
-    prefix_len = _write_prefix(newbuf, newbuf_len);
-    ret = os::vsnprintf(newbuf + prefix_len, newbuf_len - prefix_len, fmt, saved_args);
-    assert(ret >= 0, "Log message buffer issue");
-    log(level, newbuf);
-    FREE_C_HEAP_ARRAY(char, newbuf);
-  } else {
+  size_t newbuf_len = (size_t)ret + prefix_len + 1; // total bytes needed including prefix.
```

Comment:
After the assert, I would return if ret < 0 to capture the release path.
Otherwise you could end up malloc'ing a buffer which is too short by one char.

Reply:
If we return if ret < 0, we should log the buf at least.